src/librbd/cache/pwl/ssd/WriteLog.cc

```diff
@@ -708,6 +708,8 @@ bool WriteLog<I>::retire_entries(const unsigned long int frees_per_tx) {
           m_first_valid_entry = first_valid_entry;
           ceph_assert(m_first_valid_entry % MIN_WRITE_ALLOC_SSD_SIZE == 0);
           this->m_free_log_entries += retiring_entries.size();
+          ceph_assert(this->m_bytes_allocated >= allocated_bytes);
+          this->m_bytes_allocated -= allocated_bytes;
```

Comment:
This logic does exist in RWL, but it is omitted here.

Reply:
I'm not sure I understand the comment.  Do you mean that it is omitted intentionally?

Reply:
No, it's not omitted intentionally. This PR looks correct. I actually noticed this issue and fixed it locally along with a few more fixes and haven't upstreamed them yet. And I am currently running some tests as well. I will review this change once again by tomorrow. Thank you for this PR!

Reply:
FWIW I have another PR for the ssd mode almost ready -- fixing power cycle issues.  Should be able to submit by tomorrow.  I'll CC you, thanks for taking a look!